src/errors.ts

```diff
@@ -1,48 +1,166 @@
-import { inspect } from 'util';
-import { Info as InfoObject, Options, VError } from 'verror';
-
-export { InfoObject };
-export { errorForEach, errorFromList, findCauseByName, fullStack, hasCauseWithName, info as errorInfo } from 'verror';
+function createExtendableError(): { new(msg: string): Error } {
```

Comment:
Isn't this just `class ExtendableError extends Error {....}`?

And why the `function` wrapper? It is only used below as far as I can see, and from what I can see, it could simply be used with an `extends`, without the `function`.

Reply:
Unfortunately no, `Error` is a special beast that doesn't like being extended. This is the simplest way I could find to extend `Error`. If you have a (tried and tested) better way, please let me know.

Reply:
Oh, by the way, the function wrapper is for TypeScript, TS doesn't allow extending from ordinary constructor functions (only `class`es are allowed).

Reply:
Ah, the function wrapper is not necessary, found a better way.